### DISCUSSION TITLE
consider Cirrus CI?

### DISCUSSION THREAD

**Discussion by ZhiHanZ** [databendlabs/databend#1497] [Ideas]
Currently Unit Tests speed are not desired, mainly due to the hardware limitation in github action, we could consider to choose other CI platform for faster CI speed
Cirrus CI has the following advantages:
1. free for open source with generous tier(one task could use 8 cpu cores)
2. famous community such as Rust itself and flutter used this CI
3. easy to integrate with github

https://cirrus-ci.org/examples/#rust

**Comment by PsiACE** [comment]
Roughly how much improvement?

**Reply by ZhiHanZ** [reply]
github action provide only 2-core cpus, and for cirrus CI, in maximum, we could get 8-core cpus for free
https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
https://cirrus-ci.org/faq/#are-there-any-limits

**Reply by PsiACE** [reply]
I think we can try it and migrate only the unittests ci.

**Reply by sundy-li** [reply]
>  we could get 8-core cpus for free

I think it may be temporarily free,  as http://travis-ci.org/ is ceased,  do you think` cirrus CI` can provide more free accounts than Microsoft's Github?

**Comment by bohutang** [comment]
Hmm, I think we should measure it how improvement what's Cirrus gained.
Github slow but always (most time) works.

**Comment by dragonly** [comment]
This seems like a stale topic?

**Comment by sundy-li** [comment]
Are there any famous open-source database repos/stories using Cirrus CI? 
I saw pg is using this.

Sounds like we can get more free CPU using it.
![image](https://user-images.githubusercontent.com/3325189/143547761-b9903fe7-36ed-482c-94fe-b52b05219476.png)

**Comment by Xuanwo** [comment]
Introducing new CI services in projects always suffer.

How about setup github self-hosted runners if we really want to improve the CI speed? In this way, we don't need to learn different CI services' syntax. And we don't need to be afraid of meeting the hardware limitation again.

**Reply by bohutang** [reply]
The main issue is cost, how to meet our needs at a lower cost.
We plan to use self-hosted

**Reply by dragonly** [reply]
@Xuanwo @BohuTANG The cost is always the best excuse NOT to use a technology/service. Do you have any (rough) statistics about the cost? Thanks!

**Reply by bohutang** [reply]
Not yet, we like free :D
